api/app/src/main/kotlin/packit/service/BasePacketService.kt

```diff
@@ -74,4 +65,31 @@ class BasePacketService(
     {
         return this.joinToString("") { "%02x".format(it) }
     }
+
+    override fun getMetadataBy(id: String): PacketMetadata
+    {
+        return outpackServerClient.getMetadataById(id)
+            ?: throw PackitException("doesNotExist", HttpStatus.NOT_FOUND)
+    }
+
+    override fun getFileByHash(hash: String): Pair<InputStreamResource, HttpHeaders>
+    {
+        val response = outpackServerClient.getFileByHash(hash)
+
+        if (response?.first == null || response.first.isEmpty())
+        {
+            throw PackitException("doesNotExist", HttpStatus.NOT_FOUND)
+        }
+
+        val inputStream = response.first.toString().byteInputStream()
+
+        val inputStreamResource = InputStreamResource(inputStream)
+
+        val headers = HttpHeaders().apply {
+            contentType = response.second.contentType
+            contentDisposition = response.second.contentDisposition
```

Comment:
Actually maybe it _is_ appropriate for Packit to be the one controlling this. Can this endpoint accept a query parameter `inline` and if present, don't add this `contentDisposition` header? It would also be ideal if this endpoint accepted a filename (the property `path` from the file metadata) so that it can use that as the attachment header filename when present...